setup missing in CI

Tests are failing because setup is missing. See [here](https://github.com/sympy/sympy/pull/27314#issuecomment-2499037402), for example, where the error reported is

```
Run bin/test_setup.py
Traceback (most recent call last):
  File "/home/runner/work/sympy/sympy/bin/test_setup.py", line 22, in <module>
    import setup
  File "/home/runner/work/sympy/sympy/setup.py", line 33, in <module>
    from setuptools import setup, Command
ModuleNotFoundError: No module named 'setuptools'
Error: Process completed with exit code 1.
```

--- a/.github/workflows/runtests.yml
+++ b/.github/workflows/runtests.yml
@@ -21,7 +21,7 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: '3.12'
-      - run: python -m pip install --upgrade pip
+      - run: python -m pip install --upgrade pip setuptools
 
       - run: pip install -r requirements-dev.txt
       - run: pip install ruff

